sigmf-spec.md

```diff
@@ -262,6 +262,13 @@ SigMF metadata fundamentally takes the form of key/value pairs:
 "namespace:name": value,
 ```
 
+Metadata field keys in the top level `global` object, `captures` segment
+objects, or `annotations` objects MUST be of this form. Other than these top
+level objects which contain a `:` delimiter between the namespace and key name,
+field keys SHALL use only letters, numbers, and the `_` character, all other
+characters are forbidden. Additionally, field keys MUST NOT start with a number
+and keys MUST NOT not be C++ or python keywords.
```

Comment:
Only request: we should specify which releases of C++ and Python we are restricting, here.

Reply:
Per chat.gnuradio discussion will update to specify Python 3.10 and C++20 👍